packages/babel-core/src/parse.js

```diff
@@ -39,9 +39,16 @@ export const parse: Parse = (function parse(code, opts, callback) {
     opts = undefined;
   }
 
-  // For backward-compat with Babel 7's early betas, we allow sync parsing when
-  // no callback is given. Will be dropped in some future Babel major version.
-  if (callback === undefined) return parseRunner.sync(code, opts);
+  if (callback === undefined) {
+    const message =
+      "Starting from Babel 8.0.0, the 'parse' function expects a callback. If you need to call it synchronously, please use 'parseSync'.";
```

Comment:
I think it would be better to differentiate the message.
e.g.
```
Starting from Babel 8, the 'parse' function expects a callback. If you need to call it synchronously, please use 'parseSync'.
```
and
```
Starting from Babel 8, the 'parse' function will expect a callback. If you need to call it synchronously, please use 'parseSync'.
```

It would be confusing otherwise, since users will check their package.json and notice that they _aren't_ using Babel 8.